File: SingleNumber/SingleNumber.cpp
#include <stdio.h>

class Solution {
public:
  int singleNumber(int A[], int n) {
    int result = 0;
    for (int i = 0; i < n; ++i) {
      result ^= A[i];
    }
    return result;
  }
};

int main()
{
  int a[] = {1,1,2,2,3};
  Solution *s = new Solution();
  printf("%d\n", s->singleNumber(a, sizeof(a)/sizeof(a[0])));
  return 0;
}

File: SingleNumber/single_number.py
#!/usr/bin/python

'''
Given an array of integers, every element appears twice except for one. Find that single one.
'''

class Solution(object):
    def single_number(self, A):
        '''
        Given an array of integers, every element appears twice except for one. Find that single one.

        @param A, a list of integer
        @return an integer
        '''

        if not A:
            return None

        ret = 0
        for i in A:
            try:
                ret ^= i
            except TypeError:
                raise TypeError("Please input integer")
        return ret
